guides/keras_hub/semantic_segmentation_deeplab_v3.py

```diff
@@ -0,0 +1,874 @@
+# convert to 0, and [128, 0, 0] will be converted to 1, so on so forth. Also
+# note that the mask class is 1 base since class 0 is reserved for the
+# background. The [128, 0, 0] (class 1) is mapped to `aeroplane`.
+VOC_PNG_COLOR_VALUE = [
+    [0, 0, 0],
+    [128, 0, 0],
+    [0, 128, 0],
+    [128, 128, 0],
+    [0, 0, 128],
+    [128, 0, 128],
+    [0, 128, 128],
+    [128, 128, 128],
+    [64, 0, 0],
+    [192, 0, 0],
+    [64, 128, 0],
+    [192, 128, 0],
+    [64, 0, 128],
+    [192, 0, 128],
+    [64, 128, 128],
+    [192, 128, 128],
+    [0, 64, 0],
+    [128, 64, 0],
+    [0, 192, 0],
+    [128, 192, 0],
+    [0, 64, 128],
+]
+# Will be populated by _maybe_populate_voc_color_mapping() below.
+VOC_PNG_COLOR_MAPPING = None
+
+
+def _maybe_populate_voc_color_mapping():
+    # Lazy creation of VOC_PNG_COLOR_MAPPING, which could take 64M memory.
+    global VOC_PNG_COLOR_MAPPING
+    if VOC_PNG_COLOR_MAPPING is None:
+        VOC_PNG_COLOR_MAPPING = [0] * (256**3)
+        for i, colormap in enumerate(VOC_PNG_COLOR_VALUE):
+            VOC_PNG_COLOR_MAPPING[
+                (colormap[0] * 256 + colormap[1]) * 256 + colormap[2]
+            ] = i
+        # There is a special mapping with [224, 224, 192] -> 255
+        VOC_PNG_COLOR_MAPPING[224 * 256 * 256 + 224 * 256 + 192] = 255
+        VOC_PNG_COLOR_MAPPING = tf.constant(VOC_PNG_COLOR_MAPPING)
+    return VOC_PNG_COLOR_MAPPING
+
+
+def _download_data_file(
+    data_url, extracted_dir, local_dir_path=None, override_extract=False
+):
+    """Fetch the original VOC or Semantic Boundaries Dataset from remote URL.
+
+    Args:
+        data_url: string, the URL for the data to be downloaded, should be in a
+            zipped tar package.
+        local_dir_path: string, the local directory path to save the data.
+    Returns:
+        the path to the folder of extracted data.
+    """
+    if not local_dir_path:
+        # download to ~/.keras/datasets/fname
+        cache_dir = os.path.join(os.path.expanduser("~"), ".keras/datasets")
+        fname = os.path.join(os.path.basename(data_url))
+    else:
+        # Make sure the directory exists
+        if not os.path.exists(local_dir_path):
+            os.makedirs(local_dir_path, exist_ok=True)
+        # download to local_dir_path/fname
+        fname = os.path.join(os.path.basename(data_url))
+        cache_dir = local_dir_path
+    data_directory = os.path.join(os.path.dirname(fname), extracted_dir)
+    if not override_extract and os.path.exists(data_directory):
+        logging.info("data directory %s already exist", data_directory)
+        return data_directory
+    data_file_path = keras.utils.get_file(
+        fname=fname, origin=data_url, cache_dir=cache_dir
+    )
+    # Extra the data into the same directory as the tar file.
+    data_directory = os.path.dirname(data_file_path)
+    logging.info("Extract data into %s", data_directory)
+    with tarfile.open(data_file_path) as f:
```

Comment:
Can you just use `extract = True` in `get_file`? I think everything in this function is covered by `get_file`.

Reply:
Done, Thanks for the suggestion.